lib/db.js

```diff
@@ -909,6 +909,24 @@ module.exports = (
       )
   }
 
+  DB.prototype.createSigninCode = function (uid) {
+    log.trace({ op: 'DB.createSigninCode' })
+
+    return random(config.signinCodeSize)
+      .then(code => {
+        const data = { uid, createdAt: Date.now() }
+        return this.pool.put(`/signinCodes/${code.toString('hex')}`, data)
+          .then(() => code, err => {
+            if (isRecordAlreadyExistsError(err)) {
+              log.warn({ op: 'DB.createSigninCode.duplicate' })
+              return this.createSigninCode(uid)
```

Comment:
Should we limit the number of re-attempts for recreating a token? As super unlikely that it is for there to be a collision with tokens, seems strange that it could be recursive forever.